raml-validate.js

```diff
@@ -363,10 +369,19 @@ module.exports = function () {
     var validations = {}
     var validateRule = RAMLVersion === 'RAML10' ? validate.ruleRAML10 : validate.rule
 
-    // Convert all parameters into validation functions.
-    Object.keys(schema).forEach(function (param) {
-      validations[param] = validateRule(schema[param])
-    })
+    if (Array.isArray(schema.type) && schema.type.length === 1) {
+      schema.type = schema.type[0]
+    }
+
+    var isObjectType = RAML10_NON_OBJECT_TYPES.indexOf(schema.type) === -1
```

Comment:
`schema.type: 'object'` isn't set for all object types. But I probably could just default to that when `schema.type` isn't set at all.